app/script/event/EventRepository.js

```diff
@@ -611,7 +611,11 @@ z.event.EventRepository = class EventRepository {
       : Promise.resolve(event);
 
     return mapEvent
-      .then(mappedEvent => (this.eventProcessMiddleware ? this.eventProcessMiddleware(mappedEvent) : mappedEvent))
+      .then(mappedEvent => {
+        return this.eventProcessMiddlewares.reduce((eventPromise, middleware) => {
+          return eventPromise.then(middleware);
+        }, Promise.resolve(mappedEvent));
+      })
```

Comment:
Why not use `Promise.all()`?

```js
const eventPromises = this.eventProcessMiddlewares.map(eventPromise => eventPromise.then(processedEvent => middleware(processedEvent)));
return Promise.all(eventPromises);
```

Reply:
Because we need to process them sequentially, one after the other. I added a comment to explain what is happening here